platform/storage/posix/src/mender-storage.c

```diff
@@ -185,6 +186,82 @@ mender_storage_delete_deployment_data(void) {
 #ifdef CONFIG_MENDER_CLIENT_ADD_ON_CONFIGURE
 #ifdef CONFIG_MENDER_CLIENT_CONFIGURE_STORAGE
 
+mender_err_t
+mender_storage_save_update_state(mender_update_state_t state, const char *artifact_type) {
```

Comment:
I don't think this needs to be guarded by `CONFIG_MENDER_CLIENT_ADD_ON_CONFIGURE` and `CONFIG_MENDER_CLIENT_CONFIGURE_STORAGE`? It's the same in the zephyr storage, but you don't do it in the header.